### DISCUSSION TITLE
$dispatch() event in Alpine.js is called twice

### DISCUSSION THREAD

**Discussion by alikoza** [alpinejs/alpine#4205] [1. Help]
I am working on the livewire/Alpinejs project. I have an alpine blade as following :

```
<div class="tile-container" x-data="{
    tiles: Array(28).fill().map((_, idx) => idx),
  
   animate(key) {
    $dispatch('showMe', { info: key });
   }
                  
}">

<template x-for="(tile, index) in tiles" :key="index">
    <div class="tile" :id="'tile-' + index"
       x-on:click="animate(index)">
       </div>
  </template>

</div>
```

in my component I am listening the event and can see that the event has been dispatched twice at the same time. I checked the network tab and I see that the dispatch event has been fired twice :

```
[
    {
        "path": "",
        "method": "__dispatch",
        "params": [
            "showeMe",
            {
                "info": 1
            }
        ]
    },
    {
        "path": "",
        "method": "__dispatch",
        "params": [
            "showeMe",
            {
                "info": 1
            }
        ]
    }
]
```

I need this approach because I am having private variable where I need to maintain the state of this variable. if I use $wire.call then the state of private variable is not being preserved. I have tried almost all, but no success. 

I don't have multiple Alpinejs instances loaded so that is not be the reason of multiple dispatch request. I am not sure whether or not it is a bug, but one thing I know is that I can't make it work.. Any help would be highly appreciated.

**Comment by SimoTod** [comment]
Are you importing alpine in a script tag (your own bundle/cdn)? If yes, you do have multiple instances of Alpine because livewire already comes with Alpine out of the box

**Reply by alikoza** [reply]
No, I am not importing alpine in script tag. Livewire's own alpine module is being used. I wonder if it happens only in my case or it is like that in everybody ? So I have only one instances of Alpine. I also confirmed this by adding "window.Alpine" in console tab.
